File: hw/CEDENOR/U3/HW29Composite&TemplateMethod/Python/Composite/controller/Clerk.py
import sys
import os
# Nahomi Cedeño, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.Employee import Employee

class Clerk(Employee):
    def __init__(self, name):
        super().__init__(name, "Clerk")

    def add(self, employee):
        pass  # No operation for leaf nodes

    def remove(self, employee):
        pass  # No operation for leaf nodes

File: hw/CEDENOR/U3/HW29Composite&TemplateMethod/Java/CompositePattern/src/main/java/ec/edu/espe/compositepattern/controller/Clerk.java
package ec.edu.espe.compositepattern.controller;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO - ESPE
 */
public class Clerk extends Employee {
    public Clerk(String aName) {
        this();
        name = aName;
    }

    public Clerk() {
        super();
        title = "Clerk";
    }

    @Override
    public void stateName() {
        super.stateName();
    }
}
